theano/sandbox/gpuarray/tests/test_elemwise.py

```diff
@@ -46,6 +46,26 @@ def test_c_inplace(self):
         super(test_gpu_Broadcast, self).test_c_inplace()
 
 
+def test_elemwise_pow():
+    # Test that GpuElemwise(pow) can compile with any combination of integer
+    # or float input dtype.
+    dev = theano.sandbox.gpuarray.init_dev.device
+    if not dev.startswith('cuda'):
+        raise SkipTest("Cuda specific tests")
+
+    dtypes = ["uint8", "uint16", "uint32", "uint64",
+              "int8", "int16", "int32", "int64",
+              "float32", "float64"]
```

Comment:
Adding "float16" makes compilation crash without the kernels for float16 installed (the same error as with integer types before : "calling std::pow from a kernel function is forbidden and you shouldn't") so I'm not certain it's a good idea.